src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -1256,6 +1264,179 @@ void InputProcessor::getObjectDefMaxArgs(std::string const &ObjectWord, // Objec
+    DataOutputs::iTotalFieldsWithDefaults = 0;     // Total number of fields that could be defaulted
+    DataOutputs::iNumberOfAutoSizedFields = 0;     // Number of autosized fields in IDF
+    DataOutputs::iTotalAutoSizableFields = 0;      // Total number of autosizeable fields
+    DataOutputs::iNumberOfAutoCalcedFields = 0;    // Number of autocalculated fields
+    DataOutputs::iTotalAutoCalculatableFields = 0; // Total number of autocalculatable fields
+
+    auto const &schema_properties = schema.at("properties");
+
+    // Lambda to avoid repeating code twice (when processing regular fields, and extensible fields)
+    auto processField = [](const std::string& field, const json& epJSONObj, const json& schema_field_obj) {
+        bool hasDefault = false;
+        bool canBeAutosized = false;
+        bool canBeAutocalculated = false;
+
+        // If we wanted to count number of fields, would do it here
+
+        std::string defaultValue;
+
+        auto const &default_it = schema_field_obj.find("default");
+        if (default_it != schema_field_obj.end()) {
+            ++DataOutputs::iTotalFieldsWithDefaults;
+            hasDefault = true;
+            auto const &default_val = default_it.value();
+            if (default_val.is_string()) {
+                defaultValue = default_val.get<std::string>();
+            }
+        }
+
+        auto const &anyOf_it = schema_field_obj.find("anyOf");
+        if (anyOf_it != schema_field_obj.end()) {
+            for (auto const &anyOf : anyOf_it.value()) {
+                auto const &enum_it = anyOf.find("enum");
+                if (enum_it != anyOf.end()) {
+                    for (auto const &e: enum_it.value()) {
+                        if (e.is_string()) {
+                            auto const &enumVal = e.get<std::string>();
+                            if (enumVal == "Autosize") {
+                                ++DataOutputs::iTotalAutoSizableFields;
+                                canBeAutosized = true;
+                            } else if (enumVal == "Autocalculate") {
+                                ++DataOutputs::iTotalAutoCalculatableFields;
+                                canBeAutocalculated = true;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Locate the field in the ep_object
+        auto it = epJSONObj.find(field);
+        if (it != epJSONObj.end()) { // && !it.value().empty()) {
+            // Found it: check if Autosized or Autocalculated
+            auto const &field_value = it.value();
+            if (field_value.is_string()) {
+                std::string const val = field_value.get<std::string>();
+                // In the IDF, casing is an issue and Autosize/Autocalculate are accepted as synonyms (in schema it's not)
+                if (icompare(val, "Autosize") || icompare(val, "Autocalculate")) {
```

Comment:
When the IDF is converted to epJSON internally, it should use the correctly cased value in the epJSON schema. You shouldn't need to do a case-insensitive compare here (and if you do then I think that'd be a bug in the input processor).

Reply:
Thanks for pointing it out, epJSON does correctly resolve synonyms Autosize/Autocaculate and casing

in a1ea9b5 I have modified my tests & the code to avoid the extra check